.github/workflows/npm-publish-cht-deploy.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish cht-deploy to npm
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./scripts/deploy
+
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20.x'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run tests
+        run: npm test
+
+      - name: Publish to npm
+        uses: JS-DevTools/npm-publish@v3
+        with:
+          token: ${{ secrets.NPM_TOKEN }}
+          access: public
+          package: ./scripts/deploy/package.json
```

Comment:
In cht-conf we use semantic release to do this and it would be great to use the same in both places, but that library is designed to also update the version and generate release notes so maybe it's overkill. What do you think?

Reply:
Yes, semantic release is cool and more comprehensive. I liked the automated changelog.md. I was assuming that it would try to include all the CHT related commits in the release notes. That [may not be the case.](https://stackoverflow.com/questions/69342978/is-it-possible-to-exclude-some-commits-from-semantic-release-release-notes). If that's not the case, there is a [commits analyzer](https://github.com/semantic-release/commit-analyzer) to filter out commits that don't have anything to do with cht-deploy. I'll give it a look.